v0.10 changelog (#1402)
PennyLaneAI/catalyst: doc/releases/changelog-dev.md

  dictionary for the shots. For now, we still keep the shots as an attribute so device implementors 
  can have time to migrate, but we will remove shots from the attribute dictionary in the next 
  release (`v0.11`)

* The `toml` module has been migrated to PennyLane with an updated schema for declaring device
  capabilities. Devices with TOML files using `schema = 2` will not be compatible with the latest
  Catalyst. See the [Custom Devices documentation page](https://docs.pennylane.ai/projects/catalyst/en/stable/dev/custom_devices.html)
  for updated instructions on integrating your device with Catalyst and PennyLane.
  [(#1275)](https://github.com/PennyLaneAI/catalyst/pull/1275)

* Handling for the legacy operator arithmetic (the `Hamiltonian` and `Tensor` classes in PennyLane)
  has been removed.
  [(#1308)](https://github.com/PennyLaneAI/catalyst/pull/1308)

<h3>Bug fixes 🐛</h3>

* Fixed a bug introduced in 0.8 that breaks nested invocations of `qml.adjoint` and `qml.ctrl` (e.g., 
  `qml.adjoint(qml.adjoint(qml.H(0)))`).
  [(#1301)](https://github.com/PennyLaneAI/catalyst/issues/1301)

> * Fixed a bug in `debug.compile_executable` which would generate incorrect stride information for

Reviewer: ```suggestion
* Fixed a bug in :func:`~.debug.compile_executable` that would generate incorrect stride information for
```

  array arguments of the function, in particular when non-64bit datatypes are used.
  [(#1338)](https://github.com/PennyLaneAI/catalyst/pull/1338)

<h3>Internal changes ⚙️</h3>

* Catalyst no longer depends on or pins the `scipy` package. Instead, OpenBLAS is sourced directly
  from [`scipy-openblas32`](https://pypi.org/project/scipy-openblas32/) or 
  [Accelerate](https://developer.apple.com/accelerate/) is used.
  [(#1322)](https://github.com/PennyLaneAI/catalyst/pull/1322)
  [(#1328)](https://github.com/PennyLaneAI/catalyst/pull/1328)

* The `QuantumExtension` module—previously implemented with pybind11—has been removed. This module
  was not included in the distributed wheels and has been deprecated to align with our adoption of
  Python's stable ABI, which pybind11 does not support.
  [(#1187)](https://github.com/PennyLaneAI/catalyst/pull/1187)

> * Code to for using `lightning.qubit` with Catalyst has been moved from the Catalyst repository to 

Reviewer: Typo?

```suggestion
* Code for using `lightning.qubit` with Catalyst has been moved from the Catalyst repository to 
```

  lowering to OQD's backend calls. The physical parameters are read in from 
  [TOML](https://toml.io/en/) files during the `--quantum-to-ion` conversion. The TOML files are 
  assumed to exist by the pass (the paths to the TOML file locations are taken in as pass options), 
  with the intention that they are generated immediately before compilation during 
  hardware-calibration runs.

* The Catalyst IR has been extended to support literal values as opposed to SSA Values for static 
  parameters of quantum gates by adding a new gate called `StaticCustomOp` with lowering to regular 
  `CustomOp`.
  [(#1387)](https://github.com/PennyLaneAI/catalyst/pull/1387)

<h3>Documentation 📝</h3>

* A new tutorial going through how to write a new MLIR pass is available. The tutorial writes an
  empty pass that prints `hello world`. The code for the tutorial is located in
  [a separate github branch](https://github.com/PennyLaneAI/catalyst/commit/ba7b3438667963b307c07440acd6d7082f1960f3).
  [(#872)](https://github.com/PennyLaneAI/catalyst/pull/872)

* The `catalyst-cli` documentation has been updated to reflect the removal of the `func-name` option 

>   for trasnformation passes.

Reviewer: ```suggestion
  for transformation passes.
```
